consensus/core/src/dag_state.rs

```diff
@@ -840,6 +861,20 @@ impl DagState {
         std::mem::take(&mut self.unscored_committed_subdags)
     }
 
+    pub(crate) fn update_scoring_subdag<F>(&mut self, update_fn: F)
```

Comment:
nit: what was the reason to have update_scoring_subdag() and read_scoring_subdag(), instead of adding commit here and returning a ref from read_scoring_subdag()?

Reply:
> instead of adding commit here and returning a ref from read_scoring_subdag()

Sorry can you explain what you mean here by adding commit?

Reply:
I mean what is the downside of add methods like add_unscored_subdag() and clear_unscored_subdag()? Sees a bit more intuitive on what to expect but this is minor.

Reply:
I see what you mean, just a stylistic choice on my end. I thought it was cleaner this way instead of creating a wrapper for each internal function of `ScoringSubdag` in dag state. But I will change it to the way you suggested if that is more intuitive to understand as thats is more important